.circleci/continue_config.yml

```diff
@@ -50,18 +50,37 @@ commands:
 
   build_test:
     description: "build and test csharp code!"
+    parameters:
+      threshold:
+        default: 50
+        type: integer
     steps:
       - checkout
+      - run:
+          name: Get report generator
+          command: dotnet tool install dotnet-reportgenerator-globaltool --version 5.1.10 --tool-path Tools/
       - run:
           name: Restore solution dependencies and tools
           command: dotnet restore CircleCiTestFunction/CircleCiTestFunction.sln
       - run:
           name: Build solution 
           command: dotnet build CircleCiTestFunction/CircleCiTestFunction.sln
       - run:
-          name: "Running tests"
-          command: dotnet test CircleCiTestFunction/CircleCiTestFunction.sln
-
+          name: Run tests and build coverage report
+          command: |
+            # build reports even if dotnet test fails, then throw to inform Circle CI to mark the step as failed
+            dotnet test CircleCiTestFunction/CircleCiTestFunction.sln /p:Threshold=<< parameters.threshold >> /p:CollectCoverage=true /p:CoverletOutputFormat=cobertura --logger:"junit;LogFilePath=TestResults/NUnit.xml"
```

Comment:
just noting that we may or may not be using NUnit as the test framework - I think Mahdi had mentioned we might prefer XUnit, might be worth syncing/aligning on whatever that decision is

Reply:
Sure I'll check with him